common/config/schemas.go

```diff
@@ -0,0 +1,52 @@
+package config
+
+type Sync struct {
```

Comment:
Nice work.  Unmarshaling in Golang is non-trivial.  You can also embed structs inside of structs for unmarshaling.  This works just as well.